Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -4336,6 +4336,28 @@ void RenderBox::addVisualEffectOverflow()
         fragmentedFlow->addFragmentsVisualEffectOverflow(*this);
 }
 
+static LayoutBoxExtent shadowOutsetsForWritingMode(const LayoutBoxExtent& shadowExtent, WritingMode writingMode)
```

Comment:
What about adding an overload to RenderStyle::boxShadowExtent() that takes a WritingMode instead? There have been more and more uses of that idiom going in and I think it makes sense for this as well.

Reply:
is that really a computed style thing?

Reply:
> is that really a computed style thing? (though I guess RenderStyle is becoming less and less computed style but I still think it is the wrong direction).

In that case, should we look at moving the other functions that take WritingModes? 

Honestly, the least computed style part of the thing to me is `RenderStyle::boxShadowExtent()`. It would probably better if that was instead just `Style::shadowExtent(renderStyle->boxShadow())`.

Reply:
I'll address this in a followup.